### PR TITLE
Remove deprecated arguments and functions

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

This is for the 1.4 deprecations

### CODE REVIEW

File: pandas-stubs/core/groupby/groupby.pyi [hunk 1]
```diff
@@ -79,9 +79,7 @@ class GroupBy(BaseGroupBy[NDFrameT]):
     def resample(self, rule, *args, **kwargs): ...
     def rolling(self, *args, **kwargs): ...
     def expanding(self, *args, **kwargs): ...
-    def pad(self, limit: int | None = ...): ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/groupby/groupby.pyi:82]: note - we are missing the docs in pandas that says this is deprecated

File: pandas-stubs/core/groupby/groupby.pyi [hunk 2]
```diff
@@ -79,9 +79,7 @@ class GroupBy(BaseGroupBy[NDFrameT]):
...
     def ffill(self, limit: int | None = ...) -> DataFrame | Series: ...
-    def backfill(self, limit: int | None = ...) -> DataFrame | Series: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/groupby/groupby.pyi:84]: note - we are missing the docs in pandas that says this is deprecated

File: pandas-stubs/core/indexes/base.pyi [hunk 3]
```diff
@@ -156,7 +156,7 @@ class Index(IndexOpsMixin, PandasObject):
     def symmetric_difference(
         self, other: list[T1] | Index, result_name=..., sort=...
     ) -> Index: ...
-    def get_loc(self, key, method=..., tolerance=...): ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/indexes/base.pyi:159]: note - we are missing the docs in pandas that says this is deprecated

File: pandas-stubs/io/xml.pyi [hunk 4]
```diff
@@ -15,6 +15,7 @@ from pandas._typing import (
 
 def read_xml(
     path_or_buffer: FilePath | ReadBuffer[bytes] | ReadBuffer[str],
+    *,
```
Comment (Dr-Irv) [inline pandas-stubs/io/xml.pyi:18]: Not clear why you added this??

**Review by Dr-Irv — COMMENTED** [review]
some minor comments and a question

File: pandas-stubs/io/xml.pyi (hunk 4 above)
Comment (phofl) [reply pandas-stubs/io/xml.pyi:18]: We deprecated passing positional arguments to read_xml in 1.4

Deprecated passing arguments as positional for [read_xml()](https://pandas.pydata.org/pandas-docs/version/1.5.0rc0/reference/api/pandas.read_xml.html#pandas.read_xml) other than path_or_buffer ([GH45133](https://github.com/pandas-dev/pandas/issues/45133))

**Comment by phofl** [comment]
Opened https://github.com/pandas-dev/pandas/pull/48599 to add deprecate flags in the docs

File: pandas-stubs/io/xml.pyi (hunk 4 above)
Comment (Dr-Irv) [reply pandas-stubs/io/xml.pyi:18]: OK, the docs don't say anything about that, so maybe the docs should be updated as well.

**Review by Dr-Irv — APPROVED** [review]
thanks  @phofl

File: pandas-stubs/io/xml.pyi (hunk 4 above)
Comment (phofl) [reply pandas-stubs/io/xml.pyi:18]: I think we are not mentioning positional deprecations in the docs

**Pull request merged**